src/pgtkterm.c

```diff
@@ -2773,25 +2835,7 @@ pgtk_scroll_run (struct window *w, struct run *run)
 static void
 pgtk_update_begin (struct frame *f)
 {
-  if (! NILP (tip_frame) && XFRAME (tip_frame) == f
-      && ! FRAME_VISIBLE_P (f))
-    return;
-
-  if (! FRAME_CR_SURFACE (f))
-    {
-      int width = FRAME_PIXEL_WIDTH (f);
-      int height = FRAME_PIXEL_HEIGHT (f);
-
-      if (width > 0 && height > 0)
-	{
-	  block_input();
-	  FRAME_CR_SURFACE (f) = cairo_image_surface_create
-	    (CAIRO_FORMAT_ARGB32, width, height);
-	  unblock_input();
-	}
-    }
-
-  pgtk_clear_under_internal_border (f);
```

Comment:
In my code, pgtk_clear_under_internal_border was called. Where is it called instead in your code?

Reply:
It is still part of the redisplay interface, but I am not 100% sure if we do need it on every update cycle.

it seems to cover up that gtk3 emacs isn't really a gtk app and that gtk may draw something "under" the emacs buffer. 

Now that we integrate with the gtk drawing requests better I can't see any difference with it in or out.

I will need to do some more investigation to see if it does make a difference and I just am not seeing it.

Reply:
Just in case you don't know, this is internal border.
![what-is-internal-border](https://user-images.githubusercontent.com/7226237/77554907-013ee580-6efa-11ea-9b82-7c78cfe3d054.png)

pgtk_clear_under_internal_border draws this area in spite of its name.

Reply:
perferct, I do get some difference with and without that function call

Reply:
I have returned this back in,

doing this on drawing update still has some slight difference around the modebar compared to Xgtk emacs.

and it is called in a few places in xterm.c that can be later work to get the 2 systems more aligned